pkg/blobstore/batched_store_blob_access.go

```diff
@@ -78,6 +79,12 @@ func (ba *batchedStoreBlobAccess) flushLocked(ctx context.Context) {
 		return
 	}
 
+	ctx, span := trace.StartSpan(ctx, "blobstore.BatchedStoreBlobAccess.upload")
```

Comment:
It looks like many of these tracing spans don't contain a lot of additional data compared to what is communicated over the prototype of this function. Could I persuade you to create a separate TracingBlobAccess, just like how we already have MetricsBlobAccess? Then we could patch up pkg/blobstore/configuration to automatically instantiate it all over the place.